docs/standard/serialization/system-text-json/snippets/how-to-contd/csharp/CustomConverterInferredTypesToObject.cs

```diff
@@ -22,8 +22,18 @@ JsonTokenType.String when reader.TryGetDateTime(out DateTime datetime) => dateti
         public override void Write(
             Utf8JsonWriter writer,
             object objectToWrite,
-            JsonSerializerOptions options) =>
-            JsonSerializer.Serialize(writer, objectToWrite, objectToWrite.GetType(), options);
+            JsonSerializerOptions options)
+        {
+            var valueType = objectToWrite.GetType();
+            if (valueType == typeof(object))
```

Comment:
I might simplify this as follows:

```suggestion
            if (objectToWrite.GetType() is typeof(object))
```